### ISSUE TITLE
fix(external-link): HEAD 403/405 response should fall back to GET before reporting failure

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#274]
## Problem

Some servers reject HEAD requests with 403 (Forbidden) or 405 (Method Not Allowed) but respond normally to GET requests. gomarklint reports these URLs as broken even though they are reachable.

`curl` uses GET by default, so `curl https://paleoanthro.org/media/journal/content/PA20170044.pdf` returns 200 while gomarklint reports the link as unreachable.

Reported in: #147 (comment)

## Root cause

`internal/rule/external_link.go` — `performCheck`:

```go
resp, err := client.Do(req) // HEAD request
if err == nil {
    defer resp.Body.Close()
    return resp.StatusCode, nil  // returns 403/405 immediately, no GET fallback
}

// GET fallback only runs on network error, not on non-2xx HTTP responses
req.Method = "GET"
resp, err = client.Do(req)
```

The GET fallback is only triggered on a **network-level error** (connection refused, timeout, etc.). A valid HTTP response of 403 or 405 causes an immediate return without retrying as GET.

## Proposed fix

Also fall back to GET when HEAD returns 405 (Method Not Allowed) or 403 (Forbidden):

```go
resp, err := client.Do(req) // HEAD
if err == nil {
    defer resp.Body.Close()
    if resp.StatusCode == http.StatusMethodNotAllowed || resp.StatusCode == http.StatusForbidden {
        req.Method = "GET"
        resp2, err2 := client.Do(req)
        if err2 == nil {
            defer resp2.Body.Close()
            return resp2.StatusCode, nil
        }
    } else {
        return resp.StatusCode, nil
    }
}
// existing network error GET fallback remains
```

## Acceptance criteria

- [ ] A URL whose server returns 405 to HEAD but 200 to GET is reported as reachable
- [ ] A URL whose server returns 403 to HEAD but 200 to GET is reported as reachable
- [ ] A URL that returns 404 to both HEAD and GET is still reported as broken
- [ ] Existing HEAD-success paths (2xx, 3xx) are unaffected
- [ ] Unit tests added for the 405→GET and 403→GET fallback cases
